### ISSUE TITLE
mlUndeploy can delete the admin role

### ISSUE THREAD

**Issue by eaolson** [marklogic-community/ml-gradle#581]
In commissioning a new cluster, I have an Active Directory security group that will grant admin rights. So I created `/src/main/ml-config/security/roles/admin.json`:

```
{  
  "role-name":"admin",
  "description":"admin role",
  "external-name":["... DN for the admin security group ..."]
}
```

Then while testing, I was running `mlDeploy` and `mlUndeploy`, trying to get everything worked out. The last `mlUndeploy` deleted the admin role, leaving me with a MarkLogic instance with no administrators and no way to add any back. The only thing I could think to do was delete everything and recreate the cluster from scratch.

I do realize this was a situation entirely of my own making. I guess I was expecting `mlUndeploy`  to just remove the external-name property of the admin role, since that role is built in.

Is this something that's worth protecting against?

**Comment by rjrudin** [comment]
Yes, that's reasonable. There's a "do not delete" list already for servers and databases; makes sense for users and roles too.
